python-lib/spell_checker/spell_checker.py

```diff
@@ -0,0 +1,127 @@
+import dataiku
+from symspellpy.symspellpy import SymSpell, Verbosity
+from spell_checker.language_detection import *
+from spell_checker.dataframe_helpers import *
+from spell_checker.text_preprocessing import *
+
+
+class SpellChecker:
```

Comment:
A suggestion: The "words not to be corrected" (to be renamed to Custom Vocabulary) could be added to the dictionaries. The small complexity would be recomputing term frequencies for these words.

I let you evaluate if this should be in v1 or in v2. It's a nice to have.